src/jobs/jobs.controller.ts

```diff
@@ -297,150 +311,131 @@ export class JobsController {
-        throw new HttpException(
-          {
-            status: HttpStatus.BAD_REQUEST,
-            message:
-              "Unauthorized access to " +
-              datasetsNoAccess +
-              " datasets out of " +
-              datasetIds.length,
-          },
-          HttpStatus.BAD_REQUEST,
-        );
+        }
       }
-      return true;
-    } else if (jobParams.auth == JobsAuth.Authenticated) {
-      // verify that the user is authenticated
-      if (user === null) {
-        throw new HttpException(
-          {
-            status: HttpStatus.UNAUTHORIZED,
-          },
-          HttpStatus.UNAUTHORIZED,
-        );
+      if (jobConfiguration.create.auth = "#datasetPublic") {
+        datasetsWhere['isPublished'] = true;
+      } else if (jobConfiguration.create.auth = "#datasetAccess") {
+        datasetsWhere['$or'] = [
+          { ownerGroup: { $in: user.currentGroups } },
+          { accessGroups: { $in: user.currentGroups } },
+          { isPublished: true }
+        ]
+      } else if (jobConfiguration.create.auth = "#datasetOwner") {
+        datasetsWhere['ownerGroup'] = { $in: user.currentGroups }
       }
-      return true;
+      const numberOfDatasetsWithAccess = await this.datasetsService.count(datasetsWhere);
+      const datasetsNoAccess = datasetIds.length - numberOfDatasetsWithAccess.count;
+      jobInstance.datasetsValidation = (datasetsNoAccess == 0);
     }
 
-    throw new HttpException(
-      {
-        status: HttpStatus.BAD_REQUEST,
-        message: "Incorrect jobParams."
-      },
-      HttpStatus.BAD_REQUEST,
-    );
+
+    // instantiate the casl matrix for the user
+    const ability = this.caslAbilityFactory.createForUser(user);
+    // check if he/she can create this dataset
+    const canCreate =
+      ability.can(AuthOp.JobCreateAny, JobClass) ||
+      ability.can(AuthOp.JobCreateOwner, jobInstance) ||
+      ability.can(AuthOp.JobCreateConfiguration, jobInstance);
+
+    if (!canCreate) {
+      throw new ForbiddenException("Unauthorized to create this dataset");
+    }
+
+    return jobInstance;
   }
 
 
   /**
    * Send off to external service, update job in database if needed
    */
   async performJobCreateAction(jobInstance: JobClass): Promise<void> {
-    // it should return a single job configuration
-    const jobConfigs = await configuration().jobConfiguration;
-    const matchingConfig = jobConfigs.filter(
-      (j) => j.jobType == jobInstance.type,
-    );
-    if (matchingConfig.length != 1) {
-      // return error that job type does not exists
-      throw new HttpException(
-        {
-          status: HttpStatus.BAD_REQUEST,
-          message: "Invalid job type: " + jobInstance.type,
-        },
-        HttpStatus.BAD_REQUEST,
-      );
-    }
-    const jobActions = matchingConfig[0].create.actions;
-
-    for (var action of jobActions) {
+    for (var action of jobInstance.configuration?.create?.actions as Array<any>) {
```

Comment:
Will this come across potential errors that we don't catch because we are using the `?` operators?

Reply:
Yes. You are correct. 
I added the `?` operators because the typescript or linting was complaining. 
We need to review the code and make sure that is robust. I was concentrated on the functionality and left addressing  the typescript type annoying messages later

Reply:
enforce typing, remove isOptional wherever is possible, set default values for empty fields, use try-catch

Reply:
will create a separate issue for this